### ISSUE TITLE
Remove gadfly support / tests from ADBAPI

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8119]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8119|
|Type|enhancement|
|Created|2015-11-20 17:06:39Z|
|Branch|https://github.com/twisted/twisted/tree/no-gadfly-8119|

Gadfly doesn't even import on Python 2.7, we should fast-track the removal of bits for it.

<details><summary>Searchable metadata</summary>

```
trac-id__8119 8119
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_no_gadfly_8119 branches/no-gadfly-8119
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1448039199531263 1448039199531263
changetime__1449290136192714 1449290136192714
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46409]) Branching to no-gadfly-8119.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Okay, builds spun, please review.

This doesn't fall under the compatibility policy rules because Gadfly simply doesn't work on any platforms Twisted supports, being source-incompatible. Therefore, we can outright remove it -- I will send an email to the mailing list anyway.

Additionally, the second "maybe we should do this" actually shouldn't be there, but I have reactivated the first one. Of course, all of this code is unrun, but fixing it (by adding pySQLite + SQLite3 tests) will be in another ticket.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Please also remove the gadfly reference from docs twisted/docs/core/howto/rdbms.rst

Since the docs still talk about gadfly maybe this ticket should be merged with a .removal topfile

all good. thanks for the cleanup

please check my comments and merge

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46418]) Merge no-gadfly-8119: Remove Gadfly from tests.

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8119](https://github.com/twisted/twisted/issues/8119)
